src/client.rs

```diff
@@ -595,14 +593,34 @@ impl<S: Read + Write> Client<S> {
     /// Batch version of [`transaction_get`](#method.transaction_get).
     ///
     /// Takes a list of `txids` and returns a list of transactions.
-    pub fn batch_transaction_get<'t, I>(
-        &mut self,
-        txids: Vec<&Txid>,
-    ) -> Result<Vec<Transaction>, Error>
+    pub fn batch_transaction_get<'t, I>(&mut self, txids: I) -> Result<Vec<Transaction>, Error>
     where
         I: IntoIterator<Item = &'t Txid>,
     {
-        impl_batch_call!(self, txids, transaction_get)
+        let txs_string: Result<Vec<String>, Error> = impl_batch_call!(self, txids, transaction_get);
+        let mut txs = vec![];
+        for tx_string in txs_string? {
+            let tx: Transaction = deserialize(&Vec::<u8>::from_hex(&tx_string)?)?;
+            txs.push(tx)
+        }
+        Ok(txs)
+    }
+
+    /// Batch version of [`block_header`](#method.block_header).
+    ///
+    /// Takes a list of `heights` of blocks and returns a list of headers.
+    pub fn batch_block_header<'s, I>(&mut self, heights: I) -> Result<Vec<BlockHeader>, Error>
+    where
+        I: IntoIterator<Item = usize>,
+    {
+        let headers_string: Result<Vec<String>, Error> =
+            impl_batch_call!(self, heights, block_header);
+        let mut headers = vec![];
+        for header_string in headers_string? {
+            let header: BlockHeader = deserialize(&Vec::<u8>::from_hex(&header_string)?)?;
+            headers.push(header)
+        }
+        Ok(headers)
```

Comment:
since `Result` implements `FromIterator`, you can use `collect()` over a vec of results to get a result of vec (see [here](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect))

```suggestion
        headers_string?
            .into_iter()
            .map(|string| Ok(deserialize(&Vec::<u8>::from_hex(&string)?)?))
            .collect()
```

Reply:
nice, going to change